### ISSUE TITLE
Separate TCP Server Library

### ISSUE THREAD

**Issue by kidzopa** [pocoproject/poco#1101]
I am wondering if anyone has ever taken the time to break out the TCP Client and TCP Server into separate libraries. I look at the code base and I am overwhelmed there is way more there and it would be nice to have just the core of a basic client and basic server to refer to.

**Comment by aleks-f** [comment]
No, and there's no intent to do it. If you find it overwhelming to understand how [connection handler](https://github.com/pocoproject/poco/blob/develop/Net/testsuite/src/TCPServerTest.cpp#L39), [server](https://github.com/pocoproject/poco/blob/develop/Net/testsuite/src/TCPServerTest.cpp#L80) and [client](https://github.com/pocoproject/poco/blob/develop/Net/testsuite/src/TCPServerTest.cpp#L88) work, separating them in another library will not help.

**Comment by kidzopa** [comment]
The Poco framework is quite comprehensive and it covers many facets of web interface design. It represents a great body of work and is a very good framework. That said you mistake understanding for dealing with useless or unwanted clutter. I took a look at how to identify which files pertain to the tcp server  and I stopped at around forty in 3 dll's in my recollection. My point is more to organize code in functional blocks at least for core components. If I am building a server I would start with a base component while an HTTP server is available rooting about to sort out what is going on is a bit tedious.

There are several hundred files spread out over several libraries with decent slide presentations. Then you go to the samples say well I want to see how it works at the session level or someplace else in http or whatever and you dig and dig.

An opinion none the less! But thank you for telling me to look at the documentation!!

------------------------
“Tact is the ability to tell someone to go to hell in such a way that they look forward to the trip.” ― Winston S. Churchill

On Dec 25, 2015, at 3:28 PM, Aleksandar Fabijanic <notifications@github.com<mailto:notifications@github.com>> wrote:


No, and there's no intent to do it. If you find it overwhelming to understand how connection handler<https://github.com/pocoproject/poco/blob/develop/Net/testsuite/src/TCPServerTest.cpp#L39>, server<https://github.com/pocoproject/poco/blob/develop/Net/testsuite/src/TCPServerTest.cpp#L80> and client<https://github.com/pocoproject/poco/blob/develop/Net/testsuite/src/TCPServerTest.cpp#L88> work, separating them in another library will not help.
